src/stealth/stealth.js

```diff
@@ -42,20 +50,47 @@ export default function applyStealthToBrowser(browser, options) {
     const opts = _.defaults(options, DEFAULT_STEALTH_OPTIONS);
 
     const defaultContext = browser.defaultBrowserContext();
-    const contextPrototype = Object.getPrototypeOf(defaultContext);
 
-    const prevNewPage = contextPrototype.newPage;
+    const prevNewPage = defaultContext.newPage;
```

Comment:
This should solve the looping. Since we were overriding the prototype over and over and that caused the multiple stealth evaluations (apart from the iframes).

Reply:
I am afraid that this works only with the not incognito contexts Maybe we have to add something like active context or check if the context prototype was already patched than leave it and if not patch it. What do you think?

Reply:
Yeah, we could use a symbol marker as suggested here: https://github.com/JoshuaWise/better-sqlite3/pull/415#issuecomment-668472620